### PR TITLE
Replaced `parking_lot` with `std::sync`

### LINKED ISSUE #4610: Consider moving from parking_lot back to std locks
## What problem does this solve or what need does it fill?

In the past the locks in libstd were wrappers around the OS native locks like provided by pthreads. Depending on the use case the locks of parking_lot are faster. For this reason we currently use parking_lot instead. As part of the work tracked in https://github.com/rust-lang/rust/issues/93740 libstd is currently replacing OS native locks with custom implementations. The linux implementation is slightly slower than parking_lot when there is no contention, but it is faster than both parking_lot and the pthreads implementation of glibc and musl under light to extreme contention on at least two benchmarks (https://github.com/Amanieu/parking_lot/issues/338#issuecomment-1109971426 and https://github.com/rust-lang/rust/pull/95035#issuecomment-1073966631).

## What solution would you like?

We should benchmark the new locking implementation of libstd for the purposes of bevy and switch to them if they don't regress performance. Even if they don't improve 

## What alternative(s) have you considered?

Wait until https://github.com/rust-lang/rust/issues/93740 is fully done.

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Fixes #4610 

## Solution

- Replaced all instances of `parking_lot` locks with equivalents from `std::sync`. Acquiring locks within `std::sync` can fail, so `.expect("Lock Poisoned")` statements were added where required.

## Comments

In [this comment](https://github.com/bevyengine/bevy/issues/4610#issuecomment-1592407881), the lack of deadlock detection was mentioned as a potential reason to not make this change. From what I can gather, Bevy doesn't appear to be using this functionality within the engine. Unless it was expected that a Bevy consumer was expected to enable and use this functionality, it appears to be a feature lost without consequence.

Unfortunately, `cpal` and `wgpu` both still rely on `parking_lot`, leaving it in the dependency graph even after this change.

From my basic experimentation, this change doesn't appear to have any performance impacts, positive or negative. I tested this using `bevymark` with 50,000 entities and observed 20ms of frame-time before and after the change. More extensive testing with larger/real projects should probably be done.

### CODE REVIEW

**Comment by JoJoJet** [comment]
Instead of panicking when a lock gets poisoned, we could just bypass poisoning entirely by using `.unwrap_or_else(PoisonError::into_inner)`. I believe the general wisdom is that adding poisoning to `std`'s locks was a mistake, so it shouldn't be incorrect to bypass it.

**Comment by bushrat011899** [comment]
> Instead of panicking when a lock gets poisoned, we could just bypass poisoning entirely by using `.unwrap_or_else(PoisonError::into_inner)`. I believe the general wisdom is that adding poisoning to `std`'s locks was a mistake, so it shouldn't be incorrect to bypass it.

Didn't know about that! I'll make that change.

File: crates/bevy_asset/src/asset_server.rs [hunk 1]
```diff
@@ -171,13 +174,18 @@ impl AssetServer {
...
+            .loaders
+            .write()
+            .unwrap_or_else(std::sync::PoisonError::into_inner);
         let loader_index = loaders.len();
         for extension in extensions {
             if self
                 .server
                 .extension_to_loader_index
                 .write()
+                .unwrap_or_else(std::sync::PoisonError::into_inner)
```
Comment (james7132) [inline crates/bevy_asset/src/asset_server.rs:188]: This is being used enough, please add it to the use statements to reduce the number of fully qualified names used.

**Review by james7132 — APPROVED** [review]
Generally looks good to me, but probably needs to be rebased on main.

File: crates/bevy_asset/src/asset_server.rs (hunk 1 above)
Comment (bushrat011899) [reply crates/bevy_asset/src/asset_server.rs:188]: Can do!

**Review by JoJoJet — APPROVED** [review]
Seems reasonable. Even if this doesn't provide an immediate benefit over Syd's locks, the fact that it matches performance means I think it would be good to use the standard library's solution for synchronization.

Can we look into making PRs to migrate wgpu and cpal back to std's locks?

**Pull request merged**